### PR TITLE
Add a graph that maps all of the pull requests though out October

### PR DESCRIPTION
**Pull request by darrylpargeter**
added a graph using D3 for issue #40

### CODE REVIEW

**Comment by games647** [comment]
d3 isn't present in the bower.json or am I missing something?

**Comment by darrylpargeter** [comment]
it's not, I must have forgot to add the save flag, will fix that now

**Comment by darrylpargeter** [comment]
That has now been added

File: public/js/graph.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+ *                                                     *
+ ******************************************************/
+
+getData(); // start getting the graph data (kind of slow atm)
+
+
+function getData() {
+
+var data = []; // set empty gobal array
+var currentDate = new Date;
```
Comment (jenkoian) [inline public/js/graph.js]: Indent these.

**Comment by jenkoian** [comment]
Thanks @darrylpargeter this is absolutely fantastic! Few things to address then will get this merged 👍 

* Indentation - we've used 4 space tabs everywhere so far, can you update your code to use 4 spaces instead of 2? 
* I think we should have the start date be like the beginning of September and the end date be  like the env of November. We can then see easily how much of a difference hacktoberfest makes compared to other months, what do you reckon?

**Comment by jenkoian** [comment]
Screenshot for reference:

screen shot 2016-10-10 at 10 50 25

**Comment by darrylpargeter** [comment]
I Shall add the tab spacing.

I'm game for making the start date from the start of September, I will have to think of away around the rate limit (had a check and it is 30 requests per minutes for /search)

**Comment by jenkoian** [comment]
Thans @darrylpargeter for the rate limit stuff, we can probably cache this quite heavily. We only really need to get the data once a day. Am thinking we could store the etag and save it for 24 hours or something, sending if none match headers. We did do this at one point, will try and dig out the commit.

**Comment by jenkoian** [comment]
@darrylpargeter here you go, this kind of thing: https://github.com/jenkoian/hacktoberfest-checker/pull/7/files

**Comment by darrylpargeter** [comment]
I shall have a look and come up with something

**Comment by jenkoian** [comment]
@darrylpargeter if you address the tabbing we can look at the caching thing in a future PR (really keen to get this awesome feature out! :))

**Comment by darrylpargeter** [comment]
@jenkoian I shall up date the pull once I get home after work

**Comment by darrylpargeter** [comment]
Tab spacing should be fixed now

**Comment by jenkoian** [comment]
@darrylpargeter this looks *exactly* like what I was hoping for. I'm going to hold off merging though until I can spare some time to look at the caching. The search API we're using has a rate limit of 30 requests per minute, and we're calling it 31 times (if I've read it correctly) to render the graph. Not sure there are any other solutions than doing it once and then caching for 24 hours, but if there are I'm all ears. Thanks again for your wonderful PR, determined to get this up on the site soon 👍

**Comment by darrylpargeter** [comment]
Hey @jenkoian  sorry about the delay, but I was thinking it might be better to store the values in a json file (to save having to spin up a database) then to cache them every 24 hours as with caching we would still need to make the calls to the api which by the end of the month will take over 2 minutes. 
though if we save it all into a json file from the 1st September to the current date then when a user loads the site if the last date in the json file and current date don't match then we call the api to update the file then the next user won't have to call the api but just the json file and we shouldn't hit the rate limit this way.

what do you think?

**Comment by jenkoian** [comment]
Hey @darrylpargeter I did a bit of digging into the caching thing I was talking about and it seems the search endpoint doesn't support etags (guess because of the volatile nature of the response) so my original idea won't work. So, I am very much in favour of your approach! Seems like a neat solution to me 👍

**Comment by jenkoian** [comment]
Cheers for this, have updated the app for 2017 so closing this, feel free to re-submit! 😃

**Pull request closed without merging**